pkg/controller/certificates/sync.go

```diff
@@ -218,6 +228,26 @@ func (c *Controller) updateSecret(name, namespace string, cert, key []byte) (*ap
 	}
 	secret.Data[api.TLSCertKey] = cert
 	secret.Data[api.TLSPrivateKeyKey] = key
+
+	if secret.Annotations == nil {
+		secret.Annotations = make(map[string]string)
+	}
+
+	dnsnames, err := pki.DNSNamesForCertificate(crt)
+	if err != nil {
+		return nil, err
+	}
+	secret.Annotations[v1alpha1.AltNamesAnnotationKey] = strings.Join(dnsnames, ",")
+
+	cn, err := pki.CommonNameForCertificate(crt)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Sorry to go back and forth - a part of me is worried if we set the annotations based on the certificate itself, if this function is ever called with old certificate data but a new certificate resource (for some reason), it would be incorrect.

It's not a massive issue right now, but I'm concerned it's undocumented behaviour. For now, a comment on this function would probably suffice. Otherwise we should look at either reading the cert, or passing in the vars computed in the callsite to this function.